seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/ClickhouseSinkWriter.java

```diff
@@ -75,16 +87,17 @@ public void write(SeaTunnelRow element) throws IOException {
                             .indexOf(this.option.getShardMetadata().getShardKey());
             shardKey = element.getField(i);
         }
-        ClickhouseBatchStatement statement = statementMap.get(shardRouter.getShard(shardKey));
-        JdbcBatchStatementExecutor clickHouseStatement = statement.getJdbcBatchStatementExecutor();
-        IntHolder sizeHolder = statement.getIntHolder();
+        this.statement = statementMap.get(shardRouter.getShard(shardKey));
+        this.clickHouseStatement = statement.getJdbcBatchStatementExecutor();
+        this.sizeHolder = statement.getIntHolder();
         // add into batch
-        addIntoBatch(element, clickHouseStatement);
-        sizeHolder.setValue(sizeHolder.getValue() + 1);
+        addIntoBatch(element, this.clickHouseStatement);
+        this.sizeHolder.setValue(this.sizeHolder.getValue() + 1);
+        tryOpen();
```

Comment:
Move it into constructer

Reply:
This is a historical commentary Please check the latest